### PR TITLE
Drop reapers

### PR DESCRIPTION
**Pull request by soltysh**
/assign @deads2k @juanvallejo 

**Release note**:
```release-note
kubectl delete does not use reapers for removing objects anymore, but relies on server-side GC entirely
```

### CODE REVIEW

**Comment by soltysh** [comment]
Only last commit matters, b/c it waits for https://github.com/kubernetes/kubernetes/pull/59851 to merge first.

File: pkg/kubectl/cmd/apply_test.go [hunk 1]
```diff
@@ -1200,176 +1194,177 @@ func checkPatchString(t *testing.T, req *http.Request) {
...
-			}
-			for index, action := range scaleClient.Actions() {
-				if scaleClientExpected[index] != action.GetVerb() {
-					t.Errorf("unexpected API method called on a fake scale client, wanted = %s, got = %s at index = %d", scaleClientExpected[index], action.GetVerb(), index)
-				}
-			}
-		})
-	}
-}
+// TODO: fix this test
```
Comment (deads2k) [inline pkg/kubectl/cmd/apply_test.go]: um...  Yeah. you can't merge it like this.

**Comment by deads2k** [comment]
/hold

don't want to accidentally merge with tests commented out.

File: pkg/kubectl/cmd/apply_test.go (hunk 1 above)
Comment (soltysh) [reply pkg/kubectl/cmd/apply_test.go]: Duh :tong

**Comment by soltysh** [comment]
/hold cancel
Rebased and test fixed. @deads2k @juanvallejo ptal

File: pkg/kubectl/cmd/apply.go [hunk 2]
```diff
@@ -790,17 +769,15 @@ func (p *patcher) patch(current runtime.Object, modified []byte, source, namespa
 }
 
 func (p *patcher) deleteAndCreate(original runtime.Object, modified []byte, namespace, name string) ([]byte, runtime.Object, error) {
-	err := p.delete(namespace, name)
-	if err != nil {
+	if err := p.delete(namespace, name); err != nil {
 		return modified, nil, err
 	}
-	err = wait.PollImmediate(kubectl.Interval, p.timeout, func() (bool, error) {
+	if err := wait.PollImmediate(1*time.Second, p.timeout, func() (bool, error) {
```
Comment (deads2k) [inline pkg/kubectl/cmd/apply.go]: // TODO use wait

File: pkg/kubectl/cmd/delete.go [hunk 3]
```diff
@@ -301,14 +236,18 @@ func (o *DeleteOptions) DeleteResult(r *resource.Result) error {
...
 			options = metav1.NewDeleteOptions(int64(o.GracePeriod))
 		}
-		policy := metav1.DeletePropagationOrphan
+		policy := metav1.DeletePropagationForeground
+		if !o.Cascade {
+			policy = metav1.DeletePropagationOrphan
+		}
 		options.PropagationPolicy = &policy
-		return o.deleteResource(info, options)
+		// TODO: nilebox's https://github.com/kubernetes/kubernetes/pull/63695
```
Comment (deads2k) [inline pkg/kubectl/cmd/delete.go]: out of date comment

File: pkg/kubectl/cmd/drain.go [hunk 4]
```diff
@@ -594,7 +593,7 @@ func (o *DrainOptions) evictPods(pods []corev1.Pod, policyGroupVersion string, g
 				}
 			}
 			podArray := []corev1.Pod{pod}
-			_, err = o.waitForDelete(podArray, kubectl.Interval, time.Duration(math.MaxInt64), true, getPodFn)
+			_, err = o.waitForDelete(podArray, 2*time.Second, time.Duration(math.MaxInt64), true, getPodFn)
```
Comment (deads2k) [inline pkg/kubectl/cmd/drain.go]: didn't I see a one second above?

File: pkg/kubectl/scale_test.go [hunk 5]
```diff
@@ -626,3 +630,54 @@ func TestGenericScale(t *testing.T) {
 		})
 	}
 }
+
+func createFakeScaleClient(resource string, resourceName string, replicas int, errorsOnVerb map[string]*kerrors.StatusError) *fakescale.FakeScaleClient {
```
Comment (deads2k) [inline pkg/kubectl/scale_test.go]: Where is this from and used?

Comment (deads2k) [reply pkg/kubectl/scale_test.go]: > Where is this from and used?

an from delete, used in scale.

**Comment by deads2k** [comment]
nits

/lgtm

**Comment by soltysh** [comment]
cmd-test fixed, re-applying the label.

**Comment by soltysh** [comment]
/hold 
I'm going to add some debug to find the root cause, and I don't want this be merged by accident.

File: hack/make-rules/test-cmd-util.sh [hunk 6]
```diff
@@ -1478,7 +1478,7 @@ __EOF__
   kubectl delete jobs pi "${kube_flags[@]}"
   kubectl delete rs frontend "${kube_flags[@]}"
   kubectl delete rc frontend "${kube_flags[@]}"
-  kubectl delete ds bind "${kube_flags[@]}"
+  kubectl delete -f hack/testdata/rollingupdate-daemonset.yaml "${kube_flags[@]}"
```
Comment (deads2k) [inline hack/make-rules/test-cmd-util.sh]: revert

**Comment by deads2k** [comment]
nit

/lgtm

File: hack/make-rules/test-cmd-util.sh (hunk 6 above)
Comment (soltysh) [reply hack/make-rules/test-cmd-util.sh]: Reverted.

**Comment by soltysh** [comment]
Nit fixed, re-tagging.

**Comment by soltysh** [comment]
Hack in deployment reaper: https://github.com/kubernetes/kubernetes/blob/8306b0b1a4915d4b4e44f475aedf4345d592a7b7/pkg/kubectl/delete.go#L425-L429

**Pull request merged**